### DISCUSSION TITLE
[HowTo] Set individually: Plasma, Icons, Cursors

### DISCUSSION THREAD

**Discussion by noyannus** [oskarsh/Yin-Yang#191] [Show and tell]
Now that "Custom" works on scripts (Thanks, guys!!), it is possible to have a more fine-grained control. 

To keep window decorations (and more?) <strike> you *must unset "System"* in YY.</strike> you have to set the theme with the desired decoration in "Settings" _**both** for light and dark,_ and enable "System".

Note that order of commands may matter, that later ones may override settings inherited from earlier changes. I have not looked into this.

The script below even catches the Toolbar in Dolphin, but still misses something in Yakuake or Konsole (#180).

 <strike> Plasma task bar works from script in konsole, but not from YY. That §$!%&!!! path issue... (`command '['file:///home/...`) </strike> 

EDIT: better script.
EDIT2: even better script -- now catches all of konsole.

```
#! /usr/bin/sh

# ==============================================================
# "Global Theme"
plasma-apply-lookandfeel -a "org.manjaro.breath-light.desktop"

# "Plasma Style"
plasma-apply-desktoptheme "breath-light"

# "Colors"
plasma-apply-colorscheme "BreathLight"

# "Cursors"
plasma-apply-cursortheme "Nordic-cursors"

# "Icons" (full path!)
/usr/lib/plasma-changeicons "Breeze KDE-Story-Light"

# Wallpaper (also already in YY)
# plasma-apply-wallpaperimage "" # image file

# ===============================================================
# KONSOLE PROFILES

# set profiles
# "Window Color Scheme" -- sessions
SESSPROF='Breeze Light'
# "Profile" -- window decorations & toolbars
WINPROF='Breeze Light' 

# write defaults
printf "default profile: $WINPROF\n" 
printf "default color scheme: $SESSPROF\n" 
# default profile
sed -i s/DefaultProfile=.*/DefaultProfile="$WINPROF.profile"/ ~/.config/konsolerc
# default window color scheme
sed -i s/ColorScheme=.*/ColorScheme="$SESSPROF.profile"/ ~/.config/konsolerc

# running console processes
 for PROCID in $(pgrep konsole -u $USER) ; do
    printf "process: $PROCID\n"

    # windows
    for WINID in $(qdbus org.kde.konsole-$PROCID | grep /Windows/) ; do
        printf "\twindow: $WINID\tprofile: '$WINPROF\'\n"
        # set profile    
        qdbus org.kde.konsole-"$PROCID" $WINID org.kde.konsole.Window.setDefaultProfile "$WINPROF"
    done            
    
    # sessions
    for SESSID in $(qdbus org.kde.konsole-$PROCID | grep /Sessions/) ; do
        printf "\tsession: $SESSID\tprofile: '$SESSPROF\'\n"

        # set profile    
        qdbus org.kde.konsole-"$PROCID" $SESSID org.kde.konsole.Session.setProfile "$SESSPROF"
    done
done
#  ===============================================================

**Comment by OdinVex** [comment]
Would be nice for native integration, but a good option no less. :)

**Comment by noyannus** [comment]
And then there's all the[ config files](https://github.com/shalva97/kde-configuration-files). If one feels an itch to write a long litany of `sed`s... And knows a way to reload the configs.

**Comment by adrium** [comment]
I tried Yin-Yang, but then decided to write my own script. Maybe it is more suited for your needs: https://github.com/adrium/knightadjuster

**Comment by noyannus** [comment]
@adrium Thanks for the pointer. It allowed me to improve my example script above.
